src/RefractAPI.cc

```diff
@@ -423,23 +488,29 @@ namespace drafter {
         return element;
     }
 
-    refract::IElement* CategoryToRefract(const snowcrash::Element& element)
+    refract::IElement* CategoryToRefract(const SectionInfo<snowcrash::Element>& element)
     {
         refract::ArrayElement* category = new refract::ArrayElement;
         RefractElements content;
 
         category->element(SerializeKey::Category);
 
-        if (element.category == snowcrash::Element::ResourceGroupCategory) {
+        if (element.section.category == snowcrash::Element::ResourceGroupCategory) {
             category->meta[SerializeKey::Classes] = CreateArrayElement(SerializeKey::ResourceGroup);
-            category->meta[SerializeKey::Title] = refract::IElement::Create(element.attributes.name);
+            category->meta[SerializeKey::Title] = PrimitiveToRefract(MAKE_SECTION_INFO(element, attributes.name));
         }
-        else if (element.category == snowcrash::Element::DataStructureGroupCategory) {
+        else if (element.section.category == snowcrash::Element::DataStructureGroupCategory) {
             category->meta[SerializeKey::Classes] = CreateArrayElement(SerializeKey::DataStructures);
         }
 
-        snowcrash::Elements elements = element.content.elements();
-        std::transform(elements.begin(), elements.end(), std::back_inserter(content), ElementToRefract);
+        if (!element.section.content.elements().empty()) {
+            const snowcrash::SourceMap<snowcrash::Elements>& sourceMap = element.sourceMap.content.elements().collection.empty()
+                ? SectionInfo<snowcrash::Elements>::NullSourceMap()
+                : element.sourceMap.content.elements();
```

Comment:
It is not so simple as you propose.

- reference must be set in place where it is defined
-`sourceMap` is defined as const reference because:
    - `SectionInfo<snowcrash::Elements>::NullSourceMap()` returns const reference.
    - element.sourceMap.content.elements() is too const reference (because input `const SectionInfo<snowcrash::Element>& element `

There are possible solution: 
- use const pointer to set one of reference via `if/else`and later it dereference in `SectionInfoCollection` constructor
- extract if/else part into function

Which solution do you prefer? 

Please test changes that you propose before you write them into PR review!